app.js

```diff
@@ -11,72 +11,100 @@ const teamTwoSubtractButton = document.getElementById('team-two-subtract-button'
+teamTwoInput.addEventListener('click', () => {
+    teamTwoInput.value = '';
+});
 
+nameFormButton.addEventListener('click', () => {
+    const teamOne = teamOneInput.value;
+    const teamTwo = teamTwoInput.value;
+    // set the state to this data from the form
+    name1 = teamOne;
+    name2 = teamTwo;
     // reset the form values
-
+    score1 = 0;
+    score2 = 0;
     // refresh the current game element with new data by calling the appropriate function
+    refreshCurrentGameEl();
 });
 
 
 teamOneAddButton.addEventListener('click', () => {
     // increment the current state for team one's score
-    
+    score1++;
     // refresh the current game element with new data by calling the appropriate function
+    refreshCurrentGameEl(); 
+
 });
 
 teamTwoAddButton.addEventListener('click', () => {
     // increment the current state for team two's score
-
+    score2++;
     // refresh the current game element with new data by calling the appropriate function
+    refreshCurrentGameEl(); 
+
 });
 
 teamOneSubtractButton.addEventListener('click', () => {
     // decrement the current state for team one's score
-
+    score1--;
     // refresh the current game element with new data by calling the appropriate function
+    refreshCurrentGameEl(); 
+
 });
 
 teamTwoSubtractButton.addEventListener('click', () => {
     // decrement the current state for team two's score
-
+    score2--;
     // refresh the current game element with new data by calling the appropriate function
+    refreshCurrentGameEl();
 });
 
 finishGameButton.addEventListener('click', () => {
-    
     // add the current game to an array of games in state
+    
+    const currentGame = { 
+        name1: name1, 
+        name2: name2, 
+        score1: score1, 
+        score2: score2 };
     // it will be helpful to keep track of these games as objects with 4 properties, one for each piece of state we're tracking
     // for example, make an object like this: { name1: 'ducks', name2: 'bears' ,score1: 1, score2: 2 } 
     // then push it to your array in state
+    pastGames.push(currentGame);
     // (be sure to make a new object. do not declare the object in global scope and mutate it for reuse. This would cause difficult bugs)
-    
     displayAllGames();
-
     // reset the state to zero and empty strings
-    
+    name1 = '';
+    name2 = '';
+    score1 = 0;
+    score2 = 0;
+
     // refresh the current game element with new data by calling the appropriate function
+    refreshCurrentGameEl();
 });
 
 function refreshCurrentGameEl() {
     currentGameEl.textContent = '';
-
     teamOneLabel.textContent = name1;
     teamTwoLabel.textContent = name2;
-
-    // const gameEl = . . . 
-    // make a new gameEl here by calling renderGame with the approriate arguments. 
+    const gameEl = renderGame(name1, name2, score1, score2);
+    // make a new gameEl here by calling renderGame with the appropriate arguments. 
+    renderGame(pastGamesEl);
```

Comment:
This line isn't doing anything for you. On line 105 you're rendering the game with the appropriate arguments, assigning it to a variable and then appending that variable on line 113. `pastGamesEl` is an array and won't properly fire the function. Be wary of extra function calls as they can cause errors.